packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+import { getEnabledElement } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { BaseTool } from './base';
+import { getAnnotations } from '../stateManagement';
+import {
+  EventTypes,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+} from '../types';
+import { point } from '../utilities/math';
+import { Events, ToolModes, AnnotationStyleStates } from '../enums';
+import { drawCircle as drawCircleSvg } from '../drawingSvg';
+import { ToolGroupManager } from '../store';
+import { triggerAnnotationRenderForViewportIds } from './../utilities/triggerAnnotationRenderForViewportIds';
+import {
+  hideElementCursor,
+  resetElementCursor,
+} from '../cursors/elementCursor';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+import { getStyleProperty } from '../stateManagement/annotation/config/helpers';
+import { triggerAnnotationModified } from '../stateManagement/annotation/helpers/state';
+
+type CommonData = {
+  selectedIndex: number | null;
+  viewportIdsToRender: any[];
+  isEditingOpenContour: boolean;
+  mouseLocation: Types.Point2 | undefined;
+};
+
+class FreehandROISculptorTool extends BaseTool {
+  static toolName: string;
+  private _commonData?: CommonData = {
+    selectedIndex: null,
+    viewportIdsToRender: [],
+    isEditingOpenContour: false,
+    mouseLocation: undefined,
```

Comment:
Please call this canvasLocation - the problem is mouse location is not necessarily a mouse (could be touch), nor is it always in canvas coordinates - in fact I don't know what coordinate system this is in, so if it is in a different coordinate (say world), then name it appropriately.

Reply:
Updated